### PR TITLE
chore: add Preact example

### PR DESCRIPTION
**Pull request by fi3ework**
## Checklist

- [ ] Tests updated (or not required).
- [ ] Documentation updated (or not required).

### CODE REVIEW

File: examples/preact-component-bundle-false/package.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    "build": "rslib build"
+  },
+  "devDependencies": {
+    "@rsbuild/plugin-preact": "^1.2.0",
+    "@rsbuild/plugin-sass": "^1.1.1",
+    "@rslib/core": "workspace:*",
+    "preact": "^10.24.3"
+  },
+  "peerDependencies": {
+    "preact": "*"
```
Comment (Timeless0911) [inline examples/preact-component-bundle-false/package.json:17]: `*`？not an exact major limit?

File: examples/preact-component-bundle-false/rslib.config.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import { pluginPreact } from '@rsbuild/plugin-preact';
+import { pluginSass } from '@rsbuild/plugin-sass';
+import { type LibConfig, defineConfig } from '@rslib/core';
+
+const shared: LibConfig = {
+  bundle: false,
+  dts: {
```
Comment (Timeless0911) [inline examples/preact-component-bundle-false/rslib.config.ts:7]: `dts: true` is enough

Comment (fi3ework) [reply examples/preact-component-bundle-false/rslib.config.ts]: Other examples are changed as well.

File: examples/preact-component-bundle-false/package.json (hunk 1 above)
Comment (fi3ework) [reply examples/preact-component-bundle-false/package.json]: set to `^10`

**Review by Timeless0911 — APPROVED** [review]

**Review by Timeless0911 — APPROVED** [review]

**Pull request merged**